internal/controller/api/hiveextension/v1beta1/agentclusterinstall_types.go

```diff
@@ -0,0 +1,180 @@
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AgentClusterInstall represents a request to provision an agent based cluster.
+//
+// +k8s:openapi-gen=true
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type AgentClusterInstall struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AgentClusterInstallSpec   `json:"spec"`
+	Status AgentClusterInstallStatus `json:"status,omitempty"`
+}
+
+// AgentClusterInstallSpec defines the desired state of the AgentClusterInstall.
+type AgentClusterInstallSpec struct {
+
+	// ImageSetRef is a reference to a ClusterImageSet. The release image specified in the ClusterImageSet will be used
+	// to install the cluster.
+	ImageSetRef hivev1.ClusterImageSetReference `json:"imageSetRef"`
+
+	// ClusterDeploymentRef is a reference to the ClusterDeployment associated with this AgentClusterInstall.
+	ClusterDeploymentRef corev1.LocalObjectReference `json:"clusterDeploymentRef"`
+
+	// ClusterMetadata contains metadata information about the installed cluster. It should be populated once the cluster install is completed. (it can be populated sooner if desired, but Hive will not copy back to ClusterDeployment until the Installed condition goes True.
+	ClusterMetadata *hivev1.ClusterMetadata `json:"clusterMetadata,omitempty"`
+
+	// ManifestsConfigMapRef is a reference to user-provided manifests to
+	// add to or replace manifests that are generated by the installer.
+	ManifestsConfigMapRef *corev1.LocalObjectReference `json:"manifestsConfigMapRef,omitempty"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking Networking `json:"networking"`
+
+	// SSHPublicKey will be added to all cluster hosts for use in debugging.
+	// +optional
+	SSHPublicKey string `json:"sshPublicKey,omitempty"`
+
+	// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+	ProvisionRequirements ProvisionRequirements `json:"provisionRequirements"`
+
+	// ControlPlane is the configuration for the machines that comprise the
+	// control plane.
+	// +optional
+	ControlPlane *AgentMachinePool `json:"controlPlane,omitempty"`
+
+	// Compute is the configuration for the machines that comprise the
+	// compute nodes.
+	// +optional
+	Compute []AgentMachinePool `json:"compute,omitempty"`
+
+	// APIVIP is the virtual IP used to reach the OpenShift cluster's API.
+	// +optional
+	APIVIP string `json:"apiVIP,omitempty"`
+
+	// IngressVIP is the virtual IP used for cluster ingress traffic.
+	// +optional
+	IngressVIP string `json:"ingressVIP,omitempty"`
+}
+
+// AgentClusterInstallStatus defines the observed state of the AgentClusterInstall.
+type AgentClusterInstallStatus struct {
+	// Conditions includes more detailed status for the cluster install.
+	// +optional
+	Conditions []hivev1.ClusterInstallCondition `json:"conditions,omitempty"`
```

Comment:
maybe the type should be replaced to `"github.com/openshift/custom-resource-status/conditions/v1"` ?

Reply:
@dgoodwin

Reply:
Mixed opinions on this depending on who you talk to, but this matches Hive current patterns.

Reply:
There is some code duplication with helper functions, using `"github.com/openshift/custom-resource-status/conditions/v1"` can probably help with that.

Reply:
@dgoodwin do Hive plan to adopt `github.com/openshift/custom-resource-status`?

Reply:
Open to it and I will add a Jira but I dont think we can undertake it right now.

Reply:
Found this via https://issues.redhat.com/browse/HIVE-1521, cross-referencing.